snuba/datasets/errors_replacer.py

```diff
@@ -379,6 +383,7 @@ def process_delete_tag(
     all_columns: Sequence[FlattenedColumn],
     tag_column_map: Mapping[str, Mapping[str, str]],
     promoted_tags: Mapping[str, Sequence[str]],
+    state_name: Optional[ReplacerState],
```

Comment:
This is not guaranteed to accurately tell apart errors from events. It is more of a coincidence now. It is a flag used to identify the variable to rely on to save the groups in redis.
I would add another flag to this class properly named. Something like `disable_prewhere`. So these two can change independently as they are not semantically tied. There is no design reason for prewhere to be disabled when we use the ERROR state in redis.